elkodon/src/compiletests/publisher_tests.rs

```diff
@@ -0,0 +1,21 @@
+/// ```compile_fail
+/// use elkodon::prelude::*;
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// # let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
+/// #
+/// # let service = zero_copy::Service::new(&service_name)
+/// #     .publish_subscribe()
+/// #     .open_or_create::<u64>()?;
+/// #
+/// # let publisher = service.publisher().create()?;
+///
+/// let mut sample = publisher.loan()?;
+/// sample.payload_mut().write(1234);
+///
+/// publisher.send(sample)?; // should fail to compile
+///
+/// # Ok(())
+/// # }
+/// ```
+#[cfg(doctest)]
+fn sending_uninitialized_sample_fails_to_compile() {}
```

Comment:
I'm wondering if we should have a folder with compile tests or just a simple `compiletests.rs` until the file becomes too big. What do you think?